### PR TITLE
read the auth data from a file or env variable

### PR DESCRIPTION
**Pull request by krasi-georgiev**
Github actions provide the secrets as env variables and this PR extends prombench to read the json auth data directly from an env variable.

@nikita-mk 	can you please test me and let me know if we need to change anything else.

Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

### CODE REVIEW

File: pkg/provider/gke/gke.go [hunk 1]
```diff
@@ -60,12 +64,19 @@ type GKE struct {
...
+		log.Fatal("no auth provided! Need to either set the auth flag or the GOOGLE_APPLICATION_CREDENTIALS env variable")
 	}
-	cl, err := gke.NewClusterManagerClient(context.Background())
+
+	// When the auth variable points to a file
+	// put the file content in the variable.
+	if content, err := ioutil.ReadFile(c.Auth); err == nil {
+		c.Auth = string(content)
+	}
+	opts := option.WithCredentialsJSON([]byte(c.Auth))
```
Comment (nikita-mk) [inline pkg/provider/gke/gke.go:92]:
```suggestion
        if decoded, err := base64.StdEncoding.DecodeString(c.Auth); err == nil {
		c.Auth = string(decoded)
	}
	opts := option.WithCredentialsJSON([]byte(c.Auth))
```

Comment (nikita-mk) [reply pkg/provider/gke/gke.go:92]: We need to check if the string is base64 encoded and decode it if necesary.
I tested this locally and its working.

Comment (krasi-georgiev) [reply pkg/provider/gke/gke.go:92]: Actually I used the answer from https://stackoverflow.com/questions/8571501/how-to-check-whether-a-string-is-base64-encoded-or-not/20145780

to determine when the string is base64 encoded. This way the error handling is more deterministic.

Please test and suggest changes if needed.

Comment (nikita-mk) [reply pkg/provider/gke/gke.go:92]: I have tested it with Github Actions also and its working.

**Comment by krasi-georgiev** [comment]
@geekodour 	would you make a quick review and will merge if you don't see any issues with this change.

**Pull request merged**